task-generator/task_generator.go

```diff
@@ -2,26 +2,38 @@ package taskgenerator
 
 import (
 	"context"
+	"iter"
 	"time"
 
+	"github.com/Layr-Labs/eigensdk-go/chainio/txmgr"
 	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/Layr-Labs/eigensdk-go/utils"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	gethtypes "github.com/ethereum/go-ethereum/core/types"
 )
 
-type TaskGenerator struct {
+// Type is generic over the task input type
+type TaskManager[Input any] interface {
+	CreateNewTask(opts *bind.TransactOpts, input Input, quorumThresholdPercentage uint32, quorumNumbers []byte) (*gethtypes.Transaction, error)
```

Comment:
I don't like that this requires the implementor to return the `Transaction` because it limits the kind of things they could do inside of this. However, one could hack around this restriction by simply returning an empty transaction and nil error, and using a custom `TxManager` that ignores all transactions.

Reply:
It's a good catch. However, I think we can leave this change for a future change.